modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/DefInterfaceReferenceNode.java

```diff
@@ -20,34 +20,36 @@
 package org.elasticsearch.painless.ir;
 
 import org.elasticsearch.painless.ClassWriter;
-import org.elasticsearch.painless.FunctionRef;
 import org.elasticsearch.painless.MethodWriter;
 import org.elasticsearch.painless.symbol.ScopeTable;
+import org.objectweb.asm.Opcodes;
 
-public class NewArrayFuncRefNode extends ExpressionNode {
+public class DefInterfaceReferenceNode extends ReferenceNode {
 
     /* ---- begin node data ---- */
 
-    private FunctionRef funcRef;
+    private String defReferenceEncoding;
 
-    public void setFuncRef(FunctionRef funcRef) {
-        this.funcRef = funcRef;
+    public void setDefReferenceEncoding(String defReferenceEncoding) {
+        this.defReferenceEncoding = defReferenceEncoding;
     }
 
-    public FunctionRef getFuncRef() {
-        return funcRef;
+    public String getDefReferenceEncoding() {
+        return defReferenceEncoding;
     }
 
     /* ---- end node data ---- */
 
     @Override
     protected void write(ClassWriter classWriter, MethodWriter methodWriter, ScopeTable scopeTable) {
-        if (funcRef != null) {
-            methodWriter.writeDebugInfo(location);
-            methodWriter.invokeLambdaCall(funcRef);
-        } else {
-            // push a null instruction as a placeholder for future lambda instructions
-            methodWriter.push((String)null);
+        methodWriter.writeDebugInfo(location);
+
+        // place holder for
+        methodWriter.push((String)null);
+
+        for (String capture : getCaptures()) {
+            ScopeTable.Variable variable = scopeTable.getVariable(capture);
+            methodWriter.visitVarInsn(variable.getAsmType().getOpcode(Opcodes.ILOAD), variable.getSlot());
```

Comment:
`visitVarInsn` -> visit variable instance?

Reply:
an asm instruction to load a variable's value onto the stack